packages/core/src/fhir-to-cda/cda-templates/commons.ts

```diff
@@ -0,0 +1,192 @@
+    buildCodeCE({
+      code: coding.code,
+      codeSystem: coding.system,
+      codeSystemName: undefined,
+      displayName: coding.display,
+    })
+  );
+
+  const codeCV: CDACodeCV = {
+    ...baseCE,
+    originalText: codeableConcept.text,
+    translation: translations?.length ? translations : undefined,
+  };
+
+  return codeCV;
+}
+
+export function buildInstanceIdentifier({
+  root,
+  extension,
+  assigningAuthorityName,
+}: {
+  root?: string | undefined;
+  extension?: string | undefined;
+  assigningAuthorityName?: string | undefined;
+}): CDAInstanceIdentifier {
+  const identifier: CDAInstanceIdentifier = {};
+  if (root) identifier[rootAttribute] = root;
+  if (extension) identifier[extensionAttribute] = extension;
+  if (assigningAuthorityName) identifier[assigningAuthorityNameAttribute] = assigningAuthorityName;
+
+  return identifier;
+}
+
+export function buildInstanceIdentifiersFromIdentifier(
+  identifiers?: Identifier | Identifier[] | undefined
+): CDAInstanceIdentifier[] | Entry {
+  if (!identifiers) {
+    return withNullFlavor(undefined, rootAttribute);
+  }
+
+  const identifiersArray = Array.isArray(identifiers)
+    ? identifiers
+    : identifiers
+    ? [identifiers]
+    : [];
+  return identifiersArray.map(identifier =>
+    buildInstanceIdentifier({
+      root: identifier.system,
+      extension: identifier.value,
+      assigningAuthorityName: identifier.assigner?.display,
+    })
+  );
+}
+
+export function buildTelecom(telecoms: ContactPoint[] | undefined): CDATelecom[] {
+  if (!telecoms) {
+    return [];
+  }
+  return telecoms.map(telecom => ({
+    ...withoutNullFlavorObject(telecom.use, useAttribute),
+    ...withoutNullFlavorObject(telecom.value, valueAttribute),
+  }));
+}
+
+export function buildAddress(address?: Address[]): CDAAddress[] | undefined {
+  return address?.map(addr => ({
+    ...withoutNullFlavorObject(addr.use, useAttribute),
+    streetAddressLine: withoutNullFlavorString(addr.line?.join(" ")),
+    city: withoutNullFlavorString(addr.city),
+    state: withoutNullFlavorString(addr.state),
+    postalCode: withoutNullFlavorString(addr.postalCode),
+    country: withoutNullFlavorString(addr.country),
+    useablePeriod: {
+      low: withoutNullFlavorObject(addr.period?.start, valueAttribute),
+      high: withoutNullFlavorObject(addr.period?.end, valueAttribute),
+    },
+  })); // Using only first address
+}
+
+export function buildRepresentedOrganization(
+  organization: Organization
+): CDAOrganization | undefined {
+  return {
+    id: buildInstanceIdentifiersFromIdentifier(organization.identifier),
+    name: withoutNullFlavorString(organization.name),
+    telecom: buildTelecom(organization.telecom),
+    addr: buildAddress(organization.address),
+  };
+}
+
+export function formatDateToCDATimeStamp(dateString: string | undefined): string | undefined {
+  if (!dateString) {
+    return undefined;
+  }
+  const datePart = dateString.replace(/-/g, "");
```

Comment:
Should prob check if `datePart` is an empty string after trim(), and then return undefined?

Reply:
Fixed in a follow-up PR